apps/AppWithWearable/lib/pages/memories/page.dart

```diff
@@ -176,13 +164,30 @@ class _MemoriesPageState extends State<MemoriesPage> with AutomaticKeepAliveClie
             : SliverList(
                 delegate: SliverChildBuilderDelegate(
                   (context, index) {
-                    return MemoryListItem(
-                      memoryIdx: index,
-                      memory: memories[index],
-                      loadMemories: widget.refreshMemories,
+                    String date = groupedMemories.keys.elementAt(index);
```

Comment:
A column is not rendering optimized in flutter. So if you got 40 memories for 1 day, they will be rendered at once instead of making use of SliverList, which let the UI being rendered only what you are seeing.

Instead of grouping them, count the different dates with memories, and do the builder as .length + (num of dates)

Then compute the widget for each day respectively.

Reply:
Updated!